### DISCUSSION TITLE
Self signed jwts doesnt work

### DISCUSSION THREAD

**Discussion by Tarang** [supabase/supabase#1195] [Questions]
Hi

I'm wondering whether there was a change recently to JWTs. I have a self signed jwt using the secret in the supabase admin, I altered the expiration date and we had been using these for a while. I get a 401 trying to connect:

```
npm install -g wscat
wscat -c "wss://kjmwurpnrzxdbefshjhp.supabase.co/realtime/v1/websocket?apikey=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYXV0aGVudGljYXRlZCIsImlhdCI6MTYxNDQwMTM4NywiZXhwIjoxOTI5OTc3Mzg3fQ.wuRun5f2KuhrLiW9wlye5SJLMjrXB2_qu7f3YBv-K3g&vsn=1.0.0"
```

The JWT is valid, so im a bit unclear what's causing the issue.

**Comment by kiwicopple** [comment]
You can definitely self-sign JWTs, but Kong (the API gateway) has a fixed API key. Your request is probably getting blocked at the API Gateway before it even reaches the Realtime server:

https://kjmwurpnrzxdbefshjhp.supabase.co/rest/v1/?apikey=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYXV0aGVudGljYXRlZCIsImlhdCI6MTYxNDQwMTM4NywiZXhwIjoxOTI5OTc3Mzg3fQ.wuRun5f2KuhrLiW9wlye5SJLMjrXB2_qu7f3YBv-K3g

```
Kong Error
Invalid authentication credentials.
```

We're still working on customer API keys, but making progress here

**Reply by sullyinc** [reply]
Hey @kiwicopple! We're seeing the same problem, it seems the service_role key isn't getting past Kong like you mention:

```
{ message: 'Invalid authentication credentials' }
```

This is from a self-hosted local deploy with `docker-compose up` after a `supabase eject`. Using `supabase-js` from an express app. I'm hesitant to use the anonymous key for service role tasks, but the anon key does get through Kong.

I thought I saw something in the docs about "adding an API key to get past the Kong gateway", but alas, I can't find it anymore. Is it still in the docs somewhere?

**Reply by soedirgo** [reply]
Sorry @sullyinc, there was a bug in `supabase eject` that missed the `service_role` key. It should work now with v0.4.0 of the CLI.
